<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="DataSpaceR">
<title>Introduction to DataSpaceR • DataSpaceR</title>
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://docs.ropensci.org/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to DataSpaceR">
<meta property="og:description" content="DataSpaceR">
<meta property="og:image" content="https://docs.ropensci.org/DataSpaceR/logo.png">
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    <a href="https://ropensci.org" class="external-link"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
    <a class="navbar-brand me-2" href="../index.html">DataSpaceR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.7.6</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"></ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://dataspace.cavd.org">
    <span class="fa fa-globe"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/CAVDDataSpace">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/DataSpaceR">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to DataSpaceR</h1>
                        <h4 data-toc-skip class="author">Ju Yeong Kim</h4>
            
            <h4 data-toc-skip class="date">2022-06-15</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/DataSpaceR/blob/HEAD/vignettes/DataSpaceR.Rmd" class="external-link"><code>vignettes/DataSpaceR.Rmd</code></a></small>
      <div class="d-none name"><code>DataSpaceR.Rmd</code></div>
    </div>

    
    
<style>
table {
    display: block;
    max-width: 100%;
    max-height: 600px;
    overflow: scroll;
}
thead{
    position: sticky;
    top: 0px;
    background-color: #fff;
}
</style>
<p>This package provides a thin wrapper around <a href="https://cran.r-project.org/package=Rlabkey" class="external-link">Rlabkey</a> and connects to the the <a href="https://dataspace.cavd.org" class="external-link">CAVD DataSpace</a> database, making it easier to fetch datasets from specific studies.</p>
<div class="section level2">
<h2 id="configuration">Configuration<a class="anchor" aria-label="anchor" href="#configuration"></a>
</h2>
<p>First, go to <a href="https://dataspace.cavd.org" class="external-link">DataSpace</a> now and set yourself up with an account.</p>
<p>In order to connect to the CAVD DataSpace via <code>DataSpaceR</code>, you will need a <code>netrc</code> file in your home directory that will contain a <code>machine</code> name (hostname of DataSpace), and <code>login</code> and <code>password</code>. There are two ways to create a <code>netrc</code> file.</p>
<div class="section level3">
<h3 id="creating-a-netrc-file-with-writenetrc">Creating a netrc file with <code>writeNetrc</code><a class="anchor" aria-label="anchor" href="#creating-a-netrc-file-with-writenetrc"></a>
</h3>
<p>On your R console, create a <code>netrc</code> file using a function from <code>DataSpaceR</code>:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/writeNetrc.html">writeNetrc</a></span><span class="op">(</span></span>
<span>  login <span class="op">=</span> <span class="st">"yourEmail@address.com"</span>,</span>
<span>  password <span class="op">=</span> <span class="st">"yourSecretPassword"</span>,</span>
<span>  netrcFile <span class="op">=</span> <span class="st">"/your/home/directory/.netrc"</span> <span class="co"># use getNetrcPath() to get the default path</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>This will create a <code>netrc</code> file in your home directory. Make sure you have a valid login and password.</p>
</div>
<div class="section level3">
<h3 id="manually-creating-a-netrc-file">Manually creating a netrc file<a class="anchor" aria-label="anchor" href="#manually-creating-a-netrc-file"></a>
</h3>
<p><strong><em>Alternatively</em></strong>, you can manually create a netrc file.</p>
<ul>
<li>On Windows, this file should be named <code>_netrc</code>
</li>
<li>On UNIX/Mac, it should be named <code>.netrc</code>
</li>
<li>The file should be located in the user’s home directory, and the permissions on the file should be unreadable for everybody except the owner</li>
<li>To determine your home directory, run <code>Sys.getenv("HOME")</code> in R</li>
</ul>
<p>The following three lines must be included in the <code>.netrc</code> or <code>_netrc</code> file either separated by white space (spaces, tabs, or newlines) or commas. Multiple such blocks can exist in one file.</p>
<pre><code>machine dataspace.cavd.org
login myuser@domain.com
password supersecretpassword</code></pre>
<p>See <a href="https://www.labkey.org/wiki/home/Documentation/page.view?name=netrc" class="external-link">here</a> for more information about <code>netrc</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="initiate-a-connection">Initiate a connection<a class="anchor" aria-label="anchor" href="#initiate-a-connection"></a>
</h2>
<p>We’ll be looking at study <code>cvd256</code>. If you want to use a different study, change that string. You can instantiate multiple connections to different studies simultaneously.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/DataSpaceR/">DataSpaceR</a></span><span class="op">)</span></span>
<span><span class="va">con</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/connectDS.html">connectDS</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">con</span></span>
<span><span class="co">#&gt; &lt;DataSpaceConnection&gt;</span></span>
<span><span class="co">#&gt;   URL: https://dataspace.cavd.org</span></span>
<span><span class="co">#&gt;   User: jmtaylor@scharp.org</span></span>
<span><span class="co">#&gt;   Available studies: 273</span></span>
<span><span class="co">#&gt;     - 77 studies with data</span></span>
<span><span class="co">#&gt;     - 5049 subjects</span></span>
<span><span class="co">#&gt;     - 423195 data points</span></span>
<span><span class="co">#&gt;   Available groups: 3</span></span>
<span><span class="co">#&gt;   Available publications: 1530</span></span>
<span><span class="co">#&gt;     - 12 publications with data</span></span></code></pre></div>
<p>The call to <code>connectDS</code> instantiates the connection. Printing the object shows where it’s connected and the available studies.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">con</span><span class="op">$</span><span class="va">availableStudies</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="2%">
<col width="6%">
<col width="56%">
<col width="4%">
<col width="1%">
<col width="3%">
<col width="4%">
<col width="2%">
<col width="8%">
<col width="1%">
<col width="3%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">study_name</th>
<th align="left">short_name</th>
<th align="left">title</th>
<th align="left">type</th>
<th align="left">status</th>
<th align="left">stage</th>
<th align="left">species</th>
<th align="left">start_date</th>
<th align="left">strategy</th>
<th align="left">network</th>
<th align="left">data_availability</th>
<th align="left">ni_data_availability</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">cor01</td>
<td align="left">NA</td>
<td align="left">The correlate of risk targeted intervention study (CORTIS): A randomized, partially-blinded, clinical trial of isoniazid and rifapentine (3HP) therapy to prevent pulmonary tuberculosis in high-risk individuals identified by a transcriptomic correlate of risk</td>
<td align="left">Phase III</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Human</td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">GH-VAP</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">cvd232</td>
<td align="left">Parks_RV_232</td>
<td align="left">​Limiting Dose Vaginal SIVmac239 Challenge of RhCMV-SIV vaccinated Indian rhesus macaques.</td>
<td align="left">Pre-Clinical NHP</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Rhesus macaque</td>
<td align="left">2009-11-24</td>
<td align="left">Vector vaccines (viral or bacterial)</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">cvd234</td>
<td align="left">Zolla-Pazner_Mab_test1 Study</td>
<td align="left">Zolla-Pazner_Mab_Test1</td>
<td align="left">Antibody Screening</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Non-Organism Study</td>
<td align="left">2009-02-03</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">cvd235</td>
<td align="left">mAbs potency</td>
<td align="left">Weiss mAbs potency</td>
<td align="left">Antibody Screening</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Non-Organism Study</td>
<td align="left">2008-08-21</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">cvd236</td>
<td align="left">neutralization assays</td>
<td align="left">neutralization assays</td>
<td align="left">Antibody Screening</td>
<td align="left">Active</td>
<td align="left">In Progress</td>
<td align="left">Non-Organism Study</td>
<td align="left">2009-02-03</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">cvd238</td>
<td align="left">Gallo_PA_238</td>
<td align="left">HIV-1 neutralization responses in chronically infected individuals</td>
<td align="left">Antibody Screening</td>
<td align="left">Inactive</td>
<td align="left">Assays Completed</td>
<td align="left">Non-Organism Study</td>
<td align="left">2009-01-08</td>
<td align="left">Prophylactic neutralizing Ab</td>
<td align="left">CAVD</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
</tbody>
</table>
<p><code>con$availableStudies</code> shows the available studies in the CAVD DataSpace. Check out <a href="https://docs.ropensci.org/DataSpaceR/reference/DataSpaceConnection.html">the reference page</a> of <code>DataSpaceConnection</code> for all available fields and methods.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cvd256</span> <span class="op">&lt;-</span> <span class="va">con</span><span class="op">$</span><span class="fu">getStudy</span><span class="op">(</span><span class="st">"cvd256"</span><span class="op">)</span></span>
<span><span class="va">cvd256</span></span>
<span><span class="co">#&gt; &lt;DataSpaceStudy&gt;</span></span>
<span><span class="co">#&gt;   Study: cvd256</span></span>
<span><span class="co">#&gt;   URL: https://dataspace.cavd.org/CAVD/cvd256</span></span>
<span><span class="co">#&gt;   Available datasets:</span></span>
<span><span class="co">#&gt;     - Binding Ab multiplex assay</span></span>
<span><span class="co">#&gt;     - Demographics</span></span>
<span><span class="co">#&gt;     - Neutralizing antibody</span></span>
<span><span class="co">#&gt;   Available non-integrated datasets:</span></span></code></pre></div>
<p><code>con$getStudy</code> creates a connection to the study <code>cvd256</code>. Printing the object shows where it’s connected, to what study, and the available datasets.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">cvd256</span><span class="op">$</span><span class="va">availableDatasets</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">name</th>
<th align="left">label</th>
<th align="right">n</th>
<th align="left">integrated</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">BAMA</td>
<td align="left">Binding Ab multiplex assay</td>
<td align="right">6740</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Demographics</td>
<td align="left">Demographics</td>
<td align="right">121</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NAb</td>
<td align="left">Neutralizing antibody</td>
<td align="right">1419</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">cvd256</span><span class="op">$</span><span class="va">treatmentArm</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="4%">
<col width="5%">
<col width="4%">
<col width="7%">
<col width="8%">
<col width="4%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th align="left">arm_id</th>
<th align="left">arm_part</th>
<th align="left">arm_group</th>
<th align="left">arm_name</th>
<th align="left">randomization</th>
<th align="left">coded_label</th>
<th align="right">last_day</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">cvd256-NA-A-A</td>
<td align="left">NA</td>
<td align="left">A</td>
<td align="left">A</td>
<td align="left">Vaccine</td>
<td align="left">Group A Vaccine</td>
<td align="right">168</td>
<td align="left">DNA-C 4 mg administered IM at weeks 0, 4, and 8 AND NYVAC-C 10^7pfu/mL administered IM at week 24</td>
</tr>
<tr class="even">
<td align="left">cvd256-NA-B-B</td>
<td align="left">NA</td>
<td align="left">B</td>
<td align="left">B</td>
<td align="left">Vaccine</td>
<td align="left">Group B Vaccine</td>
<td align="right">168</td>
<td align="left">DNA-C 4 mg administered IM at weeks 0 and 4 AND NYVAC-C 10^7pfu/mL administered IM at weeks 20 and 24</td>
</tr>
</tbody>
</table>
<p>Available datasets and treatment arm information for the connection can be accessed by <code>availableDatasets</code> and <code>treatmentArm</code>.</p>
</div>
<div class="section level2">
<h2 id="fetching-datasets">Fetching datasets<a class="anchor" aria-label="anchor" href="#fetching-datasets"></a>
</h2>
<p>We can grab any of the datasets listed in the connection (<code>availableDatasets</code>).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NAb</span> <span class="op">&lt;-</span> <span class="va">cvd256</span><span class="op">$</span><span class="fu">getDataset</span><span class="op">(</span><span class="st">"NAb"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">NAb</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 1419   33</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">NAb</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "participant_id"      "participant_visit"   "visit_day"          </span></span>
<span><span class="co">#&gt;  [4] "assay_identifier"    "summary_level"       "specimen_type"      </span></span>
<span><span class="co">#&gt;  [7] "antigen"             "antigen_type"        "virus"              </span></span>
<span><span class="co">#&gt; [10] "virus_type"          "virus_insert_name"   "clade"              </span></span>
<span><span class="co">#&gt; [13] "neutralization_tier" "tier_clade_virus"    "target_cell"        </span></span>
<span><span class="co">#&gt; [16] "initial_dilution"    "titer_ic50"          "titer_ic80"         </span></span>
<span><span class="co">#&gt; [19] "response_call"       "nab_lab_source_key"  "lab_code"           </span></span>
<span><span class="co">#&gt; [22] "exp_assayid"         "titer_id50"          "titer_id80"         </span></span>
<span><span class="co">#&gt; [25] "nab_response_id50"   "nab_response_id80"   "slope"              </span></span>
<span><span class="co">#&gt; [28] "vaccine_matched"     "study_prot"          "virus_full_name"    </span></span>
<span><span class="co">#&gt; [31] "virus_species"       "virus_host_cell"     "virus_backbone"</span></span></code></pre></div>
<p>The <em>cvd256</em> object is an <a href="https://cran.r-project.org/package=R6" class="external-link"><code>R6</code></a> class, so it behaves like a true object. Functions (like <code>getDataset</code>) are members of the object, thus the <code>$</code> semantics to access member functions.</p>
<p>We can get detailed variable information using <code>getDatasetDescription</code>. <code>getDataset</code> and <code>getDatasetDescription</code> accept either the <code>name</code> or <code>label</code> field listed in <code>availableDatasets</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">cvd256</span><span class="op">$</span><span class="fu">getDatasetDescription</span><span class="op">(</span><span class="st">"NAb"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="17%">
<col width="8%">
<col width="66%">
</colgroup>
<thead><tr class="header">
<th align="left">fieldName</th>
<th align="left">caption</th>
<th align="left">type</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ParticipantId</td>
<td align="left">Participant ID</td>
<td align="left">Text (String)</td>
<td align="left">Subject identifier</td>
</tr>
<tr class="even">
<td align="left">antigen</td>
<td align="left">Antigen name</td>
<td align="left">Text (String)</td>
<td align="left">The name of the antigen (virus) being tested.</td>
</tr>
<tr class="odd">
<td align="left">antigen_type</td>
<td align="left">Antigen type</td>
<td align="left">Text (String)</td>
<td align="left">The standardized term for the type of virus used in the construction of the nAb antigen.</td>
</tr>
<tr class="even">
<td align="left">assay_identifier</td>
<td align="left">Assay identifier</td>
<td align="left">Text (String)</td>
<td align="left">Name identifying assay</td>
</tr>
<tr class="odd">
<td align="left">clade</td>
<td align="left">Virus clade</td>
<td align="left">Text (String)</td>
<td align="left">The clade (gene subtype) of the virus (antigen) being tested.</td>
</tr>
<tr class="even">
<td align="left">exp_assayid</td>
<td align="left">Experimental Assay Design Code</td>
<td align="left">Integer</td>
<td align="left">Unique ID assigned to the experiment design of the assay for tracking purposes.</td>
</tr>
<tr class="odd">
<td align="left">initial_dilution</td>
<td align="left">Initial dilution</td>
<td align="left">Number (Double)</td>
<td align="left">Indicates the initial specimen dilution.</td>
</tr>
<tr class="even">
<td align="left">lab_code</td>
<td align="left">Lab ID</td>
<td align="left">Text (String)</td>
<td align="left">A code indicating the lab performing the assay.</td>
</tr>
<tr class="odd">
<td align="left">nab_lab_source_key</td>
<td align="left">Data provenance</td>
<td align="left">Integer</td>
<td align="left">Details regarding the provenance of the assay results.</td>
</tr>
<tr class="even">
<td align="left">nab_response_ID50</td>
<td align="left">Response call ID50</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if neutralization is detected based on ID50 titer.</td>
</tr>
<tr class="odd">
<td align="left">nab_response_ID80</td>
<td align="left">Response call ID80</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if neutralization is detected based on ID80 titer.</td>
</tr>
<tr class="even">
<td align="left">neutralization_tier</td>
<td align="left">Neutralization tier</td>
<td align="left">Text (String)</td>
<td align="left">A classification specific to HIV NAb assay design, in which an antigen is assessed for its ease of neutralization (1=most easily neutralized, 3=least easily neutralized)</td>
</tr>
<tr class="odd">
<td align="left">response_call</td>
<td align="left">Response call</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if neutralization is detected.</td>
</tr>
<tr class="even">
<td align="left">slope</td>
<td align="left">Slope</td>
<td align="left">Number (Double)</td>
<td align="left">The slope calculated using the difference between 50% and 80% neutralization.</td>
</tr>
<tr class="odd">
<td align="left">specimen_type</td>
<td align="left">Specimen type</td>
<td align="left">Text (String)</td>
<td align="left">The type of specimen used in the assay. For nAb assays, this is generally serum or plasma.</td>
</tr>
<tr class="even">
<td align="left">study_prot</td>
<td align="left">Study Protocol</td>
<td align="left">Text (String)</td>
<td align="left">Study protocol</td>
</tr>
<tr class="odd">
<td align="left">summary_level</td>
<td align="left">Data summary level</td>
<td align="left">Text (String)</td>
<td align="left">Defines the level at which the magnitude or response has been summarized (e.g. summarized at the isolate level).</td>
</tr>
<tr class="even">
<td align="left">target_cell</td>
<td align="left">Target cell</td>
<td align="left">Text (String)</td>
<td align="left">The cell line used in the assay to determine infection (lack of neutralization). Generally TZM-bl or A3R5, but can also be other cell lines or non-engineered cells.</td>
</tr>
<tr class="odd">
<td align="left">tier_clade_virus</td>
<td align="left">Neutralization tier + Antigen clade + Virus</td>
<td align="left">Text (String)</td>
<td align="left">A combination of neutralization tier, antigen clade, and virus used for filtering.</td>
</tr>
<tr class="even">
<td align="left">titer_ID50</td>
<td align="left">Titer ID50</td>
<td align="left">Number (Double)</td>
<td align="left">The adjusted value of 50% maximal inhibitory dilution (ID50).</td>
</tr>
<tr class="odd">
<td align="left">titer_ID80</td>
<td align="left">Titer ID80</td>
<td align="left">Number (Double)</td>
<td align="left">The adjusted value of 80% maximal inhibitory dilution (ID80).</td>
</tr>
<tr class="even">
<td align="left">titer_ic50</td>
<td align="left">Titer IC50</td>
<td align="left">Number (Double)</td>
<td align="left">The half maximal inhibitory concentration (IC50).</td>
</tr>
<tr class="odd">
<td align="left">titer_ic80</td>
<td align="left">Titer IC80</td>
<td align="left">Number (Double)</td>
<td align="left">The 80% maximal inhibitory concentration (IC80).</td>
</tr>
<tr class="even">
<td align="left">vaccine_matched</td>
<td align="left">Antigen vaccine match indicator</td>
<td align="left">True/False (Boolean)</td>
<td align="left">Indicates if the interactive part of the antigen was designed to match the immunogen in the vaccine.</td>
</tr>
<tr class="odd">
<td align="left">virus</td>
<td align="left">Virus name</td>
<td align="left">Text (String)</td>
<td align="left">The term for the virus (antigen) being tested.</td>
</tr>
<tr class="even">
<td align="left">virus_backbone</td>
<td align="left">Virus backbone</td>
<td align="left">Text (String)</td>
<td align="left">Indicates the backbone used to generate the virus if from a different plasmid than the envelope.</td>
</tr>
<tr class="odd">
<td align="left">virus_full_name</td>
<td align="left">Virus full name</td>
<td align="left">Text (String)</td>
<td align="left">The full name of the virus used in the construction of the nAb antigen.</td>
</tr>
<tr class="even">
<td align="left">virus_host_cell</td>
<td align="left">Virus host cell</td>
<td align="left">Text (String)</td>
<td align="left">The host cell used to incubate the virus stock.</td>
</tr>
<tr class="odd">
<td align="left">virus_insert_name</td>
<td align="left">Virus insert name</td>
<td align="left">Text (String)</td>
<td align="left">The amino acid sequence inserted in the virus construct.</td>
</tr>
<tr class="even">
<td align="left">virus_species</td>
<td align="left">Virus species</td>
<td align="left">Text (String)</td>
<td align="left">A classification for virus species using informal taxonomy.</td>
</tr>
<tr class="odd">
<td align="left">virus_type</td>
<td align="left">Virus type</td>
<td align="left">Text (String)</td>
<td align="left">The type of virus used in the construction of the nAb antigen.</td>
</tr>
<tr class="even">
<td align="left">visit_day</td>
<td align="left">Visit Day</td>
<td align="left">Integer</td>
<td align="left">Target study day defined for a study visit. Study days are relative to Day 0, where Day 0 is typically defined as enrollment and/or first injection.</td>
</tr>
</tbody>
</table>
<p>To get only a subset of the data and speed up the download, filters can be passed to <code>getDataset</code>. The filters are created using the <code>makeFilter</code> function of the <code>Rlabkey</code> package.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cvd256Filter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Rlabkey/man/makeFilter.html" class="external-link">makeFilter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"visit_day"</span>, <span class="st">"EQUAL"</span>, <span class="st">"0"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">NAb_day0</span> <span class="op">&lt;-</span> <span class="va">cvd256</span><span class="op">$</span><span class="fu">getDataset</span><span class="op">(</span><span class="st">"NAb"</span>, colFilter <span class="op">=</span> <span class="va">cvd256Filter</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">NAb_day0</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 709  33</span></span></code></pre></div>
<p>See <code><a href="https://rdrr.io/pkg/Rlabkey/man/makeFilter.html" class="external-link">?makeFilter</a></code> for more information on the syntax.</p>
</div>
<div class="section level2">
<h2 id="creating-a-connection-to-all-studies">Creating a connection to all studies<a class="anchor" aria-label="anchor" href="#creating-a-connection-to-all-studies"></a>
</h2>
<p>To fetch data from multiple studies, create a connection at the project level.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cavd</span> <span class="op">&lt;-</span> <span class="va">con</span><span class="op">$</span><span class="fu">getStudy</span><span class="op">(</span><span class="st">""</span><span class="op">)</span></span></code></pre></div>
<p>This will instantiate a connection at the <code>CAVD</code> level. Most functions work cross study connections just like they do on single studies.</p>
<p>You can get a list of datasets available across all studies.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cavd</span></span>
<span><span class="co">#&gt; &lt;DataSpaceStudy&gt;</span></span>
<span><span class="co">#&gt;   Study: CAVD</span></span>
<span><span class="co">#&gt;   URL: https://dataspace.cavd.org/CAVD</span></span>
<span><span class="co">#&gt;   Available datasets:</span></span>
<span><span class="co">#&gt;     - Binding Ab multiplex assay</span></span>
<span><span class="co">#&gt;     - Demographics</span></span>
<span><span class="co">#&gt;     - Enzyme-Linked ImmunoSpot</span></span>
<span><span class="co">#&gt;     - Intracellular Cytokine Staining</span></span>
<span><span class="co">#&gt;     - Neutralizing antibody</span></span>
<span><span class="co">#&gt;     - PK MAb</span></span>
<span><span class="co">#&gt;   Available non-integrated datasets:</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">cavd</span><span class="op">$</span><span class="va">availableDatasets</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">name</th>
<th align="left">label</th>
<th align="right">n</th>
<th align="left">integrated</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">BAMA</td>
<td align="left">Binding Ab multiplex assay</td>
<td align="right">170320</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">Demographics</td>
<td align="left">Demographics</td>
<td align="right">5049</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">ELISPOT</td>
<td align="left">Enzyme-Linked ImmunoSpot</td>
<td align="right">5610</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">ICS</td>
<td align="left">Intracellular Cytokine Staining</td>
<td align="right">195883</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">NAb</td>
<td align="left">Neutralizing antibody</td>
<td align="right">51382</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">PKMAb</td>
<td align="left">PK MAb</td>
<td align="right">3217</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<p>In all-study connection, <code>getDataset</code> will combine the requested datasets. Note that in most cases, the datasets will have too many subjects for quick data transfer, making filtering of the data a necessity. The <code>colFilter</code> argument can be used here, as described in the <code>getDataset</code> section.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">conFilter</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Rlabkey/man/makeFilter.html" class="external-link">makeFilter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"species"</span>, <span class="st">"EQUAL"</span>, <span class="st">"Human"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">human</span> <span class="op">&lt;-</span> <span class="va">cavd</span><span class="op">$</span><span class="fu">getDataset</span><span class="op">(</span><span class="st">"Demographics"</span>, colFilter <span class="op">=</span> <span class="va">conFilter</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 3142   36</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">human</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "subject_id"                      "subject_visit"                  </span></span>
<span><span class="co">#&gt;  [3] "species"                         "subspecies"                     </span></span>
<span><span class="co">#&gt;  [5] "sexatbirth"                      "race"                           </span></span>
<span><span class="co">#&gt;  [7] "ethnicity"                       "country_enrollment"             </span></span>
<span><span class="co">#&gt;  [9] "circumcised_enrollment"          "bmi_enrollment"                 </span></span>
<span><span class="co">#&gt; [11] "agegroup_range"                  "agegroup_enrollment"            </span></span>
<span><span class="co">#&gt; [13] "age_enrollment"                  "study_label"                    </span></span>
<span><span class="co">#&gt; [15] "study_start_date"                "study_first_enr_date"           </span></span>
<span><span class="co">#&gt; [17] "study_fu_complete_date"          "study_public_date"              </span></span>
<span><span class="co">#&gt; [19] "study_network"                   "study_last_vaccination_day"     </span></span>
<span><span class="co">#&gt; [21] "study_type"                      "study_part"                     </span></span>
<span><span class="co">#&gt; [23] "study_group"                     "study_arm"                      </span></span>
<span><span class="co">#&gt; [25] "study_arm_summary"               "study_arm_coded_label"          </span></span>
<span><span class="co">#&gt; [27] "study_randomization"             "study_product_class_combination"</span></span>
<span><span class="co">#&gt; [29] "study_product_combination"       "study_short_name"               </span></span>
<span><span class="co">#&gt; [31] "study_grant_pi_name"             "study_strategy"                 </span></span>
<span><span class="co">#&gt; [33] "study_prot"                      "genderidentity"                 </span></span>
<span><span class="co">#&gt; [35] "studycohort"                     "bmi_category"</span></span></code></pre></div>
<p>Check out <a href="https://docs.ropensci.org/DataSpaceR/reference/DataSpaceStudy.html">the reference page</a> of <code>DataSpaceStudy</code> for all available fields and methods.</p>
</div>
<div class="section level2">
<h2 id="connect-to-a-saved-group">Connect to a saved group<a class="anchor" aria-label="anchor" href="#connect-to-a-saved-group"></a>
</h2>
<p>A group is a curated collection of participants from filtering of treatments, products, studies, or species, and it is created in <a href="https://dataspace.cavd.org/cds/CAVD/app.view" class="external-link">the DataSpace App</a>.</p>
<p>Let’s say you are using the App to filter and visualize data and want to save them for later or explore in R with <code>DataSpaceR</code>. You can save a group by clicking the Save button on the Active Filter Panel.</p>
<p>We can browse available the saved groups or the curated groups by DataSpace Team via <code>availableGroups</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">con</span><span class="op">$</span><span class="va">availableGroups</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="13%">
<col width="13%">
<col width="48%">
<col width="4%">
<col width="2%">
<col width="1%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="right">group_id</th>
<th align="left">label</th>
<th align="left">original_label</th>
<th align="left">description</th>
<th align="left">created_by</th>
<th align="left">shared</th>
<th align="right">n</th>
<th align="left">studies</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">220</td>
<td align="left">NYVAC durability comparison</td>
<td align="left">NYVAC_durability</td>
<td align="left">Compare durability in 4 NHP studies using NYVAC-C (vP2010) and NYVAC-KC-gp140 (ZM96) products.</td>
<td align="left">ehenrich</td>
<td align="left">TRUE</td>
<td align="right">78</td>
<td align="left">cvd281, cvd434, cvd259, cvd277</td>
</tr>
<tr class="even">
<td align="right">228</td>
<td align="left">HVTN 505 case control subjects</td>
<td align="left">HVTN 505 case control subjects</td>
<td align="left">Participants from HVTN 505 included in the case-control analysis</td>
<td align="left">drienna</td>
<td align="left">TRUE</td>
<td align="right">189</td>
<td align="left">vtn505</td>
</tr>
<tr class="odd">
<td align="right">230</td>
<td align="left">HVTN 505 polyfunctionality vs BAMA</td>
<td align="left">HVTN 505 polyfunctionality vs BAMA</td>
<td align="left">Compares ICS polyfunctionality (CD8+, Any Env) to BAMA mfi-delta (single Env antigen) in the HVTN 505 case control cohort</td>
<td align="left">drienna</td>
<td align="left">TRUE</td>
<td align="right">170</td>
<td align="left">vtn505</td>
</tr>
</tbody>
</table>
<p>To fetch data from a saved group, create a connection at the project level with a group ID. For example, we can connect to the “NYVAC durability comparison” group which has group ID 220 by <code>getGroup</code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nyvac</span> <span class="op">&lt;-</span> <span class="va">con</span><span class="op">$</span><span class="fu">getGroup</span><span class="op">(</span><span class="fl">220</span><span class="op">)</span></span>
<span><span class="va">nyvac</span></span>
<span><span class="co">#&gt; &lt;DataSpaceStudy&gt;</span></span>
<span><span class="co">#&gt;   Group: NYVAC durability comparison</span></span>
<span><span class="co">#&gt;   URL: https://dataspace.cavd.org/CAVD</span></span>
<span><span class="co">#&gt;   Available datasets:</span></span>
<span><span class="co">#&gt;     - Binding Ab multiplex assay</span></span>
<span><span class="co">#&gt;     - Demographics</span></span>
<span><span class="co">#&gt;     - Enzyme-Linked ImmunoSpot</span></span>
<span><span class="co">#&gt;     - Intracellular Cytokine Staining</span></span>
<span><span class="co">#&gt;     - Neutralizing antibody</span></span>
<span><span class="co">#&gt;   Available non-integrated datasets:</span></span></code></pre></div>
<p>Retrieving a dataset is the same as before.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NAb_nyvac</span> <span class="op">&lt;-</span> <span class="va">nyvac</span><span class="op">$</span><span class="fu">getDataset</span><span class="op">(</span><span class="st">"NAb"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">NAb_nyvac</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 4281   33</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="access-virus-metadata">Access Virus Metadata<a class="anchor" aria-label="anchor" href="#access-virus-metadata"></a>
</h2>
<p>DataSpace maintains metadata about all viruses used in Neutralizing Antibody (NAb) assays. This data can be accessed through the app on the <a href="https://dataspace.cavd.org/cds/CAVD/app.view#learn/learn/Assay/NAB/antigens" class="external-link">NAb antigen page</a> and <a href="https://dataspace.cavd.org/cds/CAVD/app.view#learn/learn/Assay/NAB%20MAB/antigens" class="external-link">NAb MAb antigen page</a>.</p>
<p>We can access this metadata in <code>DataSpaceR</code> with <code>con$virusMetadata</code>:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">con</span><span class="op">$</span><span class="va">virusMetadata</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="7%">
<col width="6%">
<col width="6%">
<col width="7%">
<col width="9%">
<col width="2%">
<col width="7%">
<col width="14%">
<col width="7%">
<col width="6%">
<col width="7%">
<col width="7%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left">assay_identifier</th>
<th align="left">cds_virus_id</th>
<th align="left">virus</th>
<th align="left">virus_type</th>
<th align="left">neutralization_tier</th>
<th align="left">clade</th>
<th align="left">antigen_control</th>
<th align="left">virus_full_name</th>
<th align="left">virus_name_other</th>
<th align="left">virus_species</th>
<th align="left">virus_host_cell</th>
<th align="left">virus_backbone</th>
<th align="left">panel_names</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">NAB MAB</td>
<td align="left">cds_1</td>
<td align="left">0013095-2.11</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">NA</td>
<td align="left">0</td>
<td align="left">0013095-2.11 [SG3Δenv] 293T/17</td>
<td align="left">NA</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
<tr class="even">
<td align="left">NAB MAB</td>
<td align="left">cds_2</td>
<td align="left">001428-2.42</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">001428-2.42 [SG3Δenv] 293T/17</td>
<td align="left">NA</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
<tr class="odd">
<td align="left">NAB MAB</td>
<td align="left">cds_3</td>
<td align="left">0041.v3.c18</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">0041.v3.c18 [SG3Δenv] 293T/17</td>
<td align="left">0041.V3.C18</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">NAB MAB</td>
<td align="left">cds_4</td>
<td align="left">0077.v1.c16</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">0077.v1.c16 [SG3Δenv] 293T/17</td>
<td align="left">0077.v1.c16</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">NAB</td>
<td align="left">cds_252</td>
<td align="left">00836-2.5</td>
<td align="left">Env Pseudotype</td>
<td align="left">1B</td>
<td align="left">C</td>
<td align="left">0</td>
<td align="left">00836-2.5 [SG3Δenv] 293T/17</td>
<td align="left">NA</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
<tr class="even">
<td align="left">NAB MAB</td>
<td align="left">cds_5</td>
<td align="left">0260.v5.c1</td>
<td align="left">Env Pseudotype</td>
<td align="left">2</td>
<td align="left">A</td>
<td align="left">0</td>
<td align="left">0260.v5.c1 [SG3Δenv] 293T/17</td>
<td align="left">0260.V5.C1</td>
<td align="left">HIV</td>
<td align="left">293T/17</td>
<td align="left">SG3Δenv</td>
<td align="left">Tiered diverse panel</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="access-monoclonal-antibody-data">Access monoclonal antibody data<a class="anchor" aria-label="anchor" href="#access-monoclonal-antibody-data"></a>
</h2>
<p>See other vignette for a tutorial on accessing monoclonal antibody data with <code>DataSpaceR</code>:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Monoconal_Antibody_Data"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="browse-and-download-publication-data">Browse and Download Publication Data<a class="anchor" aria-label="anchor" href="#browse-and-download-publication-data"></a>
</h2>
<p>DataSpace maintains a curated collection of relevant publications, which can be accessed through the <a href="https://dataspace.cavd.org/cds/CAVD/app.view?#learn/learn/Publication" class="external-link">Publications page</a> through the app. Metadata about these publications can be accessed through <code>DataSpaceR</code> with <code>con$availablePublications</code>.</p>
<p>See Publication Data vignette for a tutorial on accessing publication data through DataSpaceR.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Publication_Data"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: vignette 'Publication_Data' not found</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reference-tables">Reference Tables<a class="anchor" aria-label="anchor" href="#reference-tables"></a>
</h2>
<p>The followings are the tables of all fields and methods that work on <code>DataSpaceConnection</code> and <code>DataSpaceStudy</code> objects and could be used as a quick reference.</p>
<div class="section level3">
<h3 id="dataspaceconnection">
<code>DataSpaceConnection</code><a class="anchor" aria-label="anchor" href="#dataspaceconnection"></a>
</h3>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>availableStudies</code></td>
<td>The table of available studies.</td>
</tr>
<tr class="even">
<td><code>availableGroups</code></td>
<td>The table of available groups.</td>
</tr>
<tr class="odd">
<td><code>availablePublications</code></td>
<td>The table of available publications.</td>
</tr>
<tr class="even">
<td><code>mabGrid</code></td>
<td>The filtered mAb grid.</td>
</tr>
<tr class="odd">
<td><code>mabGridSummary</code></td>
<td>The summarized mAb grid with updated <code>n_</code> columns and <code>geometric_mean_curve_ic50</code>.</td>
</tr>
<tr class="even">
<td><code>virusMetadata</code></td>
<td>Metadata about all viruses in the DataSpace.</td>
</tr>
<tr class="odd">
<td><code>filterMabGrid</code></td>
<td>Filter rows in the mAb grid by specifying the values to keep in the columns found in the <code>mabGrid</code> field.</td>
</tr>
<tr class="even">
<td><code>resetMabGrid</code></td>
<td>Reset the mAb grid to the unfiltered state.</td>
</tr>
<tr class="odd">
<td><code>getMab</code></td>
<td>Create a <code>DataSpaceMab</code> object by filtered <code>mabGrid</code>.</td>
</tr>
<tr class="even">
<td><code>getStudy</code></td>
<td>Create a <code>DataSpaceStudy</code> object by study.</td>
</tr>
<tr class="odd">
<td><code>getGroup</code></td>
<td>Create a <code>DataSpaceStudy</code> object by group.</td>
</tr>
<tr class="even">
<td><code>downloadPublicationData</code></td>
<td>Download data from a chosen publication.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="dataspacestudy">
<code>DataSpaceStudy</code><a class="anchor" aria-label="anchor" href="#dataspacestudy"></a>
</h3>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>study</code></td>
<td>The study name.</td>
</tr>
<tr class="even">
<td><code>group</code></td>
<td>The group name.</td>
</tr>
<tr class="odd">
<td><code>availableDatasets</code></td>
<td>The table of datasets available in the study object.</td>
</tr>
<tr class="even">
<td><code>treatmentArm</code></td>
<td>The table of treatment arm information for the connected study. Not available for all study connection.</td>
</tr>
<tr class="odd">
<td><code>dataDir</code></td>
<td>The default target directory for downloading non-integrated datasets.</td>
</tr>
<tr class="even">
<td><code>studyInfo</code></td>
<td>Stores the information about the study.</td>
</tr>
<tr class="odd">
<td><code>getDataset</code></td>
<td>Get a dataset from the connection.</td>
</tr>
<tr class="even">
<td><code>getDatasetDescription</code></td>
<td>Get variable information.</td>
</tr>
<tr class="odd">
<td><code>setDataDir</code></td>
<td>Set default target directory for downloading non-integrated datasets.</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="dataspacemab">
<code>DataSpaceMab</code><a class="anchor" aria-label="anchor" href="#dataspacemab"></a>
</h3>
<table class="table">
<colgroup>
<col width="50%">
<col width="50%">
</colgroup>
<thead><tr class="header">
<th>Name</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>studyAndMabs</code></td>
<td>The table of available mAbs by study.</td>
</tr>
<tr class="even">
<td><code>mabs</code></td>
<td>The table of available mAbs and their attributes.</td>
</tr>
<tr class="odd">
<td><code>nabMab</code></td>
<td>The table of mAbs and their neutralizing measurements against viruses.</td>
</tr>
<tr class="even">
<td><code>studies</code></td>
<td>The table of available studies.</td>
</tr>
<tr class="odd">
<td><code>assays</code></td>
<td>The table of assay status by study.</td>
</tr>
<tr class="even">
<td><code>variableDefinitions</code></td>
<td>The table of variable definitions.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.1.2 (2021-11-01)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 18.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.utf8       LC_NUMERIC=C             </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.utf8        LC_COLLATE=en_US.utf8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.utf8    LC_MESSAGES=en_US.utf8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.utf8       LC_NAME=C                </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C              LC_TELEPHONE=C           </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.utf8 LC_IDENTIFICATION=C      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt; [1] data.table_1.14.2 DataSpaceR_0.7.5  knitr_1.37       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] Rcpp_1.0.8       digest_0.6.29    assertthat_0.2.1 R6_2.5.1        </span></span>
<span><span class="co">#&gt;  [5] jsonlite_1.8.0   magrittr_2.0.2   evaluate_0.15    highr_0.9       </span></span>
<span><span class="co">#&gt;  [9] httr_1.4.2       stringi_1.7.6    curl_4.3.2       tools_4.1.2     </span></span>
<span><span class="co">#&gt; [13] stringr_1.4.0    Rlabkey_2.8.3    xfun_0.29        compiler_4.1.2</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start top-4 bottom-8">
            <div class="col-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="https://github.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-github"></div></a>
                        <a href="https://github.com/ropenscilabs" target="_blank" class="external-link"><div class="icon fa fa-flask"></div></a>
                        <a href="https://hachyderm.io/@ropensci" target="_blank" class="external-link"><div class="icon fab fa-mastodon"></div></a>
                        <a href="https://vimeo.com/ropensci" target="_blank" class="external-link"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about" class="external-link">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review" class="external-link">Software Review</a></li>
                            <li><a href="https://ropensci.org/about#team" class="external-link">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers" class="external-link">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate" class="external-link">Donate</a></li>
                            <li><a href="https://ropensci.org/contact" class="external-link">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/" class="external-link">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/" class="external-link">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/" class="external-link">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/" class="external-link">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct" class="external-link">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/" class="external-link">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/" class="external-link">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/" class="external-link">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/" class="external-link">Documentation</a></li>
                            <li><a href="https://ropensci.org/news/" class="external-link">Newsletter</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/" class="external-link">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" class="external-link">NumFOCUS</a>.</p>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->


    </footer>
</div>

  

  

  </body>
</html>
